pkg/kubelet/lifecycle/handlers.go

```diff
@@ -105,6 +107,31 @@ func resolvePort(portReference intstr.IntOrString, container *v1.Container) (int
 	return -1, fmt.Errorf("couldn't find port: %v in %v", portReference, container)
 }
 
+// formatURL formats a URL from args.
+func formatURL(scheme string, host string, port int, path string) *url.URL {
+	// split path into path and query
+	pathQuery := strings.SplitN(path, "?", 2)
+	u := &url.URL{
+		Scheme: scheme,
+		Host:   net.JoinHostPort(host, strconv.Itoa(port)),
+		Path:   pathQuery[0],
```

Comment:
are there tests for path containing valid and invalid %-escaped characters? does this change behavior?